documentstore_migracao/utils/convert_html_body.py

```diff
@@ -229,70 +230,294 @@ def matched_first_two_words(text_words, search_expr):
+    @staticmethod
+    def normalize(s):
+        if s:
+            return " ".join([c.strip() for c in s.strip().split() if c.strip()])
+
+    @property
+    def texts(self):
+        _texts = []
+        for node in self.tree.findall(".//*"):
+            for _text in [node.text, node.tail]:
+                _text = self.normalize(_text)
+                if _text:
+                    _texts.append(_text)
+        return _texts
+
+    @property
+    def body(self):
+        return " ".join(self.texts)
+
+    @property
+    def words(self):
+        w = []
+        for text in self.texts:
+            w.extend(text.split())
+        return w
+
+    @property
+    def text(self):
+        words = []
+        for node in self.tree.findall(".//*"):
+            for _text in [node.text, node.tail]:
+                if (_text or "").strip():
+                    words.extend([w.strip() for w in _text.split()])
+        return "".join(words)
+
+
+class DataDiffer:
+    def __init__(self, normalize_data_to_compare=None):
+        self._before = None
+        self._after = None
+        self.normalize_data_to_compare = (
+            normalize_data_to_compare or self._normalize_data_to_compare
+        )
+
+    def _normalize_data_to_compare(self, data):
+        return data
+
+    @property
+    def before(self):
+        return self._before
+
+    @before.setter
+    def before(self, data):
+        self._before = self.normalize_data_to_compare(data)
+
+    @property
+    def after(self):
+        return self._after
+
+    @after.setter
+    def after(self, data):
+        self._after = self.normalize_data_to_compare(data)
+
+    @property
+    def diff(self):
+        if type(self._after) == type(self._before):
+            return not self._after == self._before
+        raise UnableToCompareError("Unable to compare")
+
+    @property
+    def similarity_ratio(self):
+        if not type(self._after) == type(self._before):
+            raise UnableToCompareError("Unable to compare")
+        try:
+            return difflib.SequenceMatcher(
+                None, self._before, self._after).ratio()
+        except TypeError:
+            return difflib.SequenceMatcher(
+                None, sorted(self._before), sorted(self._after)).ratio()
+
+    @property
+    def info(self):
+        return {
+            "before length": len(self.before),
+            "after length": len(self.after),
+            "similarity ratio": self.similarity_ratio,
+        }
+
+
+def get_words_to_compare(data):
+    return set(XMLTexts(data).words)
+
+
+def get_body_to_compare(data):
+    return XMLTexts(data).body
+
+
+def get_text_to_compare(data):
+    return XMLTexts(data).text
+
+
+class ConversionPipe(plumber.Pipe):
+    def __init__(self, body_info):
+        super().__init__()
+        self.pipe = type(self).__name__
+        self.body_info = body_info
+        self.spy = body_info.spy
+
+    def _begin(self, data):
+        logger.debug("INICIO: %s", self.pipe)
```

Comment:
O log de "início do Pipe" acaba ficando mais acoplado e exige que os pipe sejam filhas de conversionPipe. 

Talvez deixar o log de início em cada pipe seja ou criar uma classe mais genérica em que todos os Pipes sejam filhas e herdem as características comuns. 

Não é bloqueante para aprovação, somente melhorias de arquitetura.

Reply:
@jamilatta vc não compreendeu o conteúdo de `self.pipe`. Por favor, revise.

Reply:
`self.pipe` é uma str. Mudei o nome para `self.pipe_name`.